core/controlplane/config/templates/stack-template.json

```diff
@@ -257,15 +242,37 @@
                   "Resource": "*",
                   "Effect": "Allow"
                 }
+          ]
+        }
+      }
+    },
+    "IAMRoleController": {
+      "Properties": {
+        "AssumeRolePolicyDocument": {
+          "Statement": [
+            {
+              "Action": [
+                "sts:AssumeRole"
               ],
-              "Version": "2012-10-17"
-            },
-            "PolicyName": "root"
-          }
-        ]
+              "Effect": "Allow",
+              "Principal": {
+                "Service": [
+                  "ec2.{{.Region.PublicDomainName}}"
+                ]
+              }
+            }
+          ],
+          "Version": "2012-10-17"
+        },
+        "Path": "/",
+        {{if .Controller.IAMConfig.ManagedIamRoleName }}
+        "RoleName":  {"Fn::Join": ["",[{"Ref": "AWS::Region"},"-","{{.Controller.IAMConfig.ManagedIamRoleName}}"]]},
+        {{end}}
+        "ManagedPolicyArns": [ {"Ref": "IAMManagedPolicyController"} ]
```

Comment:
i can change the config block, so instead of being

```
iam:
   managedIamRoleName
   existingInstanceProfile
```

could be:

```
   iam:
       role: 
          iamRoleName: "your role"
          managedPolicyArns:
             - "arn:aws:iam:YOURACCOUNTID/policy/POLICYID"
      instanceProfile:
          arn:  "arn:aws:iam::YOURACCOUNTID:instance-profile/INSTANCEPROFILEID"
```
that would be better @mumoshu ?

Regarding #340 i now realize is half implemented yes, if we agree on that syntax i can adapt the PR.

Thanks for your comments @mumoshu :smiley: